pkg/deploy/reaper/reaper.go

```diff
@@ -36,34 +35,24 @@ type DeploymentConfigReaper struct {
 // zero replicas, waits for all of them to get deleted and then deletes both the
 // replication controller and its deployment configuration.
 func (reaper *DeploymentConfigReaper) Stop(namespace, name string, gracePeriod *kapi.DeleteOptions) (string, error) {
-	dc, err := reaper.osc.DeploymentConfigs(namespace).Get(name)
-	if err != nil {
+	if err := reaper.osc.DeploymentConfigs(namespace).Delete(name); err != nil {
```

Comment:
This will need changed for reentrancy (the patch predates https://github.com/openshift/origin/issues/2721#issuecomment-108661712). Will fix in a followup to make it more robust. @kargakis, maybe you can pick that up if you get here first.